File: src/main/java/ro/nicolaemariusghergu/queryit/service/PromotionService.java
package ro.nicolaemariusghergu.queryit.service;

import org.springframework.http.ResponseEntity;
import ro.nicolaemariusghergu.queryit.dto.PromotionDto;

import java.util.List;

public interface PromotionService {

    ResponseEntity<PromotionDto> getPromotionById(Long id);

    ResponseEntity<List<PromotionDto>> getPromotions();

    ResponseEntity<List<PromotionDto>> getPromotionsByName(String name);

    ResponseEntity<PromotionDto> updatePromotion(PromotionDto promotionDto);

    ResponseEntity<Long> addPromotion(PromotionDto promotionDto);

    ResponseEntity<Long> deletePromotionById(Long id);

}

File: src/main/kotlin/ro/nicolaemariusghergu/queryit/service/PromotionService.kt
package ro.nicolaemariusghergu.queryit.service

import org.springframework.http.ResponseEntity
import ro.nicolaemariusghergu.queryit.dto.PromotionDto

interface PromotionService {
    fun getPromotions(): ResponseEntity<MutableList<PromotionDto>>
    fun deletePromotionById(id: Long): ResponseEntity<Long>
}